File: backend/src/main/java/ropold/backend/model/EmployeeModel.java
package ropold.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "employees")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class EmployeeModel {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(name = "personnel_number", unique = true)
    private String personnelNumber;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "email", unique = true)
    private String email;

    @Column(name = "phone", unique = true)
    private String phone;

    @Column(name = "address", columnDefinition = "text")
    private String address;

    @Enumerated(EnumType.STRING)
    @Column(name = "department", nullable = false)
    private Department department;

    @Column(name = "active", nullable = false)
    private boolean active = true;

    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;

    @Column(name = "image_url")
    private String imageUrl;
}

File: frontend/src/components/models/EmployeeModel.ts

export type Department = "MARKETING" | "ACCOUNTING" | "HR" | "DEVELOPMENT" | "MANAGEMENT";

export type EmployeeModel = {
    id: string;
    personnelNumber: string | null;
    name: string;
    email: string | null;
    phone: string | null;
    address: string | null;
    department: Department;
    active: boolean;
    notes: string | null;
    imageUrl: string | null;
};

export const DefaultEmployee: EmployeeModel = {
    id: "",
    personnelNumber: null,
    name: "",
    email: null,
    phone: null,
    address: null,
    department: "MARKETING",
    active: true,
    notes: null,
    imageUrl: null,
};